lib/matplotlib/hatch.py

```diff
@@ -181,6 +183,20 @@ def __init__(self, hatch, density):
     ]
 
 
+def _validate_hatch_pattern(hatch):
+    valid_hatch_patterns = set(r'-+|/\xXoO.*')
```

Comment:
I often find it helpful that `plt.style.available_styles` lists all the valid styles, so wonder if it worth exposing the list of valid hatch patterns somewhere? This would avoid having to re-define the list of styles in the Hatch style reference too.

Reply:
This goes a bit in the direction proposed by @brunobeltran of writing classes for our custom types like hatches or linestyles. They could expose the allowed values (if it's a simple finite list) and also carry the validator.

But that's beyond this PR.